core/src/main/java/com/linecorp/armeria/common/util/Sampler.java

```diff
@@ -42,45 +42,52 @@
  *
  * <p>Forked from brave-core 5.6.3 at 3be55b5cccf881104bdd80c93e97d2575b83952d
  */
-// abstract for factory-method support on Java language level 7
-public abstract class Sampler {
-
-    public static final Sampler ALWAYS_SAMPLE = new Sampler() {
-        @Override
-        public boolean isSampled() {
-            return true;
-        }
-
-        @Override public String toString() {
-            return "AlwaysSample";
-        }
-    };
+@FunctionalInterface
+public interface Sampler<T> {
+    /**
+     * Returns a sampler, given a rate expressed as a floating point number between {@code 0.0} and {@code 1.0}.
+     *
+     * @param rate minimum sampling rate is between {@code 0.01} and {@code 1.0}.
+     */
+    static <T> Sampler<T> random(double rate) {
+        @SuppressWarnings("unchecked")
+        final Sampler<T> cast = CountingSampler.create(rate);
+        return cast;
+    }
 
-    public static final Sampler NEVER_SAMPLE = new Sampler() {
-        @Override
-        public boolean isSampled() {
-            return false;
-        }
+    /**
+     * Returns a sampler, given a rate-limited on a per-second interval.
+     *
+     * @param samplesPerSecond an integer between {@code 0} and {@value Integer#MAX_VALUE}
+     */
+    static <T> Sampler<T> rateLimited(int samplesPerSecond) {
+        @SuppressWarnings("unchecked")
+        final Sampler<T> cast = RateLimitingSampler.create(samplesPerSecond);
+        return cast;
+    }
 
-        @Override public String toString() {
-            return "NeverSample";
-        }
-    };
+    /**
+     * Returns a sampler that will always be sampled.
+     */
+    static <T> Sampler<T> always() {
+        @SuppressWarnings("unchecked")
+        final Sampler<T> cast = Samplers.ALWAYS;
+        return cast;
+    }
 
     /**
-     *  Returns true if a request should be recorded.
+     * Returns a sampler that will never be sampled.
```

Comment:
`never be sampled` -> `always return {@code false}` ?

Reply:
Thank you for review. Fixed in 02ab62f.